services/provider/server/server_test.go

```diff
@@ -0,0 +1,298 @@
+				},
+				{
+					Name: "cephuser-rbd-node",
+					Kind: "CephClient",
+				},
+				{
+					Name: "cephuser-cephfs-provisioner",
+					Kind: "CephClient",
+				},
+				{
+					Name: "cephuser-cephfs-node",
+					Kind: "CephClient",
+				},
+				{
+					Name: "cephuser-health-checker",
+					Kind: "CephClient",
+				},
+			},
+		},
+	}
+
+	for _, i := range consumerResource.Status.CephResources {
+		if i.Kind == "CephClient" {
+			cephClient, secret := createCephClientAndSecret(i.Name, server)
+			cephClient.Status = &rookCephv1.CephClientStatus{
+				Info: map[string]string{
+					"secretName": fmt.Sprintf("rook-ceph-client-%s", i.Name),
+				},
+			}
+
+			assert.NoError(t, client.Create(ctx, cephClient))
+			assert.NoError(t, client.Create(ctx, secret))
+		}
+	}
+
+	monCm, monSc, mgrSvc := createMonConfigMapandSecret(server)
+	assert.NoError(t, client.Create(ctx, monCm))
+	assert.NoError(t, client.Create(ctx, monSc))
+	assert.NoError(t, client.Create(ctx, mgrSvc))
+
+	exR, err := server.getExternalResources(ctx, consumerResource)
+	assert.NoError(t, err)
+	for i := range mockExtR {
+		data, err := json.Marshal(mockExtR[i].Data)
+		assert.NoError(t, err)
+		assert.Equal(t, string(exR[i].Data), string(data))
+		assert.Equal(t, exR[i].Name, mockExtR[i].Name)
+		assert.Equal(t, exR[i].Kind, mockExtR[i].Kind)
+	}
+
+	// When CephClient status is empty
+
+	object = []runtime.Object{
+		&rookCephv1.CephClient{},
+	}
+	s = runtime.NewScheme()
+	assert.NoError(t, v1.AddToScheme(s))
+	client = newFakeClient(t, object...)
+	server = &OCSProviderServer{
+		client:    client,
+		namespace: "openshift-storage",
+	}
+
+	for _, i := range consumerResource.Status.CephResources {
+		if i.Kind == "CephClient" {
+			cephClient, secret := createCephClientAndSecret(i.Name, server)
+			assert.NoError(t, client.Create(ctx, cephClient))
+			assert.NoError(t, client.Create(ctx, secret))
+
+		}
+	}
+
+	exR, err = server.getExternalResources(ctx, consumerResource)
+	assert.Error(t, err)
+	assert.NotEqual(t, len(mockExtR), len(exR))
+
+	// When CephClient status info is empty
+
+	object = []runtime.Object{
+		&rookCephv1.CephClient{},
+	}
+	s = runtime.NewScheme()
+	assert.NoError(t, v1.AddToScheme(s))
+	client = newFakeClient(t, object...)
+	server = &OCSProviderServer{
+		client:    client,
+		namespace: "openshift-storage",
+	}
+
+	for _, i := range consumerResource.Status.CephResources {
+		if i.Kind == "CephClient" {
+			cephClient, secret := createCephClientAndSecret(i.Name, server)
+			cephClient.Status = &rookCephv1.CephClientStatus{
+				Info: map[string]string{},
+			}
+
+			assert.NoError(t, client.Create(ctx, cephClient))
+			assert.NoError(t, client.Create(ctx, secret))
+
+		}
+	}
+
+	monCm, monSc, mgrSvc = createMonConfigMapandSecret(server)
+	assert.NoError(t, client.Create(ctx, monCm))
+	assert.NoError(t, client.Create(ctx, monSc))
+	assert.NoError(t, client.Create(ctx, mgrSvc))
+
+	exR, err = server.getExternalResources(ctx, consumerResource)
```

Comment:
You should the public entry point (`GetStorageConfig`) not a private util function `getExternalResources`